### PR TITLE
Added API for users.

### PR DESCRIPTION
**Pull request by Myztiq**
Added users API
- [x] @thejsj
- [x] @anandkumarpatel

### CODE REVIEW

File: lib/users.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  /**
+   * Helper method to wrap all the intercomClient.users calls around
+   * `canUseIntercom`
+   * @param {String} method Name of the method to wrap for the intercom
+   *   users client.
+   * @returns {Promise} - Returns success if intercom is disabled, otherwise
+   *   returns results of call to intercom client
+   * @private
+   */
+  _wrap (method) {
```
Comment (Myztiq) [inline lib/users.js:30]: If we had another library I think we should abstract this out into a base class. But for now I think it's alright.

Comment (anandkumarpatel) [inline lib/users.js:30]: reminds me of TJ magic :)

File: lib/users.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    return Promise.resolve()
```
Comment (anandkumarpatel) [inline lib/users.js:31]: nit: Promise.try()

File: lib/users.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      })
+  }
+
+  /**
+   * Upsert a user object, passes data through to intercom
+   * @param {Object} userParams
+   * @param {Object} userParams.user_id - User unique identifier
+   * @param {Object} userParams.name - User name
+   * @returns {Promise} - Returns success if intercom is disabled, otherwise
+   *   returns results of create command in intercom
```
Comment (anandkumarpatel) [inline lib/users.js:60]: hmm should you defined what this resolves too? Like the object I should expect and its properties, or is it a string? or NaN? the world will never know

**Comment by anandkumarpatel** [comment]
looking good!

File: lib/users.js (hunk 2 above)
Comment (thejsj) [inline lib/users.js:31]: Minor: I think `Promise.try` might be a nicer way of doing this.

**Pull request merged**